qa-tests-backend/src/test/groovy/ReconciliationTest.groovy

```diff
@@ -102,58 +109,61 @@ class ReconciliationTest extends BaseSpecification {
         def namespaceID = orchestrator.createNamespace(ns)
         NamespaceService.waitForNamespace(namespaceID, 10)
 
-        // Wait is builtin
-        def secretID = orchestrator.createSecret("testing123", ns)
-        SecretService.waitForSecret(secretID, 10)
-
-        Deployment dep = new Deployment()
-                .setNamespace(ns)
-                .setName ("testing123")
-                .setImage ("quay.io/rhacs-eng/qa:busybox")
-                .addPort (22)
-                .addLabel ("app", "testing123")
-                .setCommand(["sleep", "600"])
-
-        // Wait is builtin
-        orchestrator.createDeployment(dep)
-        assert Services.waitForDeployment(dep)
-        assert Services.getPods().findAll { it.deploymentId == dep.getDeploymentUid() }.size() == 1
-
-        def violations = getViolationsWithTimeout("testing123",
-                "Secure Shell (ssh) Port Exposed", 30)
-        assert violations.size() == 1
-
-        NetworkPolicy policy = new NetworkPolicy("do-nothing")
-                .setNamespace(ns)
-                .addPodSelector()
-                .addPolicyType(NetworkPolicyTypes.INGRESS)
-        def networkPolicyID = orchestrator.applyNetworkPolicy(policy)
-        assert NetworkPolicyService.waitForNetworkPolicy(networkPolicyID)
-
-        def sensorDeployment = new Deployment().setNamespace("stackrox").setName("sensor")
-        orchestrator.deleteAndWaitForDeploymentDeletion(sensorDeployment)
-
-        def labels = ["app":"sensor"]
-        orchestrator.waitForAllPodsToBeRemoved("stackrox", labels)
-
-        orchestrator.identity {
-            // Delete objects from k8s
-            deleteDeployment(dep)
-            deleteSecret("testing123", ns)
-            deleteNetworkPolicy(policy)
-            deleteNamespace(ns)
-            // Just wait for the namespace to be deleted which is indicative that all of them have been deleted
-            waitForNamespaceDeletion(ns)
-
-            // Recreate sensor
-            try {
-                createOrchestratorDeployment(sensor)
-            } catch (Exception e) {
-                log.error("Error re-creating the sensor: ", e)
-                throw e
+        try {
+            addStackroxImagePullSecret(ns)
```

Comment:
@gavin-stackrox I did not go the way you suggested in https://issues.redhat.com/browse/ROX-11005?focusedCommentId=20297961&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20297961 - via adding `provisionDefaultServiceAccount()` to `createNamespace()` because it seems that `createNamespace()` is used in standalone fashion here https://github.com/stackrox/stackrox/blob/e0f50e9699fa1bb98604729c59e70fa1d64a5fc7/qa-tests-backend/src/test/groovy/BaseSpecification.groovy#L86-L89

I'm a bit too worried if what `provisionDefaultServiceAccount()` does could mess things up there. I don't understand the full logic of that function and afraid that testing cycle can be long, therefore trying to make quick-and-dirty fix here.

Reply:
Yeah. I think that is better bet. (createNamespace is also used in a few other tests).